blocks/bricks/__init__.py

```diff
@@ -345,15 +345,18 @@ def __init__(self, input_dim, output_dim, num_pieces, **kwargs):
         self.output_dim = output_dim
         self.num_pieces = num_pieces
 
-        self.linear_transformation = Linear(
-            name=self.name + '_linear_to_maxout', input_dim=input_dim,
-            output_dim=output_dim * num_pieces, weights_init=self.weights_init,
-            biases_init=self.biases_init, use_bias=self.use_bias)
-        self.maxout_transformation = Maxout(name=self.name + '_maxout',
-                                            num_pieces=num_pieces)
+        self.linear_transformation = Linear(name=self.name +
```

Comment:
Sure. Now we're commenting on this code anyway, I never know what to do here: Don't configure the child brick at all, and only do it in `_push_allocation_config`, or do it in both places. The advantage if only doing it in `_push_allocation_config` is that there is less code duplication. Plus, I think it makes it clearer to the user that the child brick isn't configured until `push_allocation_config` was called (if the user changes anything before that, it will just be overridden).

The advantage of doing it in both places is that sometimes you might want to delegate properties to child bricks, and you will want these to work before `push_allocation_config` was called. For example, `MLP.input_dim` should really return `MLP.children[0].input_dim` (it doesn't now, but it probably should). For that to work as early as possible, it would have to configure its first and last child's dimensions before `push_allocation_config` was called.